hypervisor/src/hypervisor/config.rs

```diff
@@ -1,3 +1,52 @@
+//! Configuration for a653rs-linux-hypervisor.
+//!
+//! A configuration contains information about the partition schedule.
+//! Currently, only a single schedule is supported. Each schedule's major frame
+//! (MAF) has a fixed duration and number of slots. Each slot is occupied by
+//! at most one partition that executes during the slot. A slot has a fixed
+//! duration and offset inside the MAF. A partition may occupy multiple slots
+//! inside the schedule, in which case it may be repeated using the `period`
+//! parameter.
+//!
+//! The hypervisor runs the executable file specified by `image` for each
+//! partition as a long-running process that is started and stopped according to
+//! the partition schedule.
+//!
+//! Partitions can communicate using channels (Sampling and Queuing). The name
+//! of the ports by which a partition can access a channel is the same for all
+//! attached partitions.
+//!
+//! ```rust
+//! use a653rs_linux_hypervisor::hypervisor::config::Config;
+//!
+//! let yaml = "
```

Comment:
Unnecessary parts of a doc test could be hidden

https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#hiding-portions-of-the-example